### PR TITLE
check: Check if safe.directory is already configured before adding an entry

### PR DESCRIPTION
**Pull request by lyarwood**
**What this PR does / why we need it**:

`git config --global --add` will always add a new line to `safe.directory` when called so check if it's actually required before calling.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: scripts/check.sh [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #!/bin/bash
 
 # We need to add the current directory as a safe git directory before running any commands
-git config --global --add safe.directory "$(pwd)"
+git config --global --get-all safe.directory | grep "$(pwd)" > /dev/null
+safedir=$?
+if [[ $safedir -ne 0 ]]; then
```
Comment (0xFelix) [inline scripts/check.sh:6]: Introducing the `$safedir` variable is not necessary, you can check on `$?` directly or if you want to make it even more compact this should work too:

```
if ! git config --global --get-all safe.directory | grep "$(pwd)" > /dev/null; then
    git config --global --add safe.directory "$(pwd)"
fi
```

Comment (lyarwood) [reply scripts/check.sh:6]: I was sure I tried this at the time but was getting weird results with this command, I'll try again.

Comment (lyarwood) [reply scripts/check.sh]: Huh that works now, I think I may have wrapped it accidentally with `$()` before for some reason.

**Pull request merged**